### ISSUE TITLE
Why mochi diffusers video output is worse than mochi official code?

### ISSUE THREAD

**Issue by foreverpiano** [huggingface/diffusers#10144]
### Describe the bug

The quality of video is worse.

### Reproduction

Run the code with official prompt

### System Info

diffusers@main

### Who can help?

@a-r-r-o-w @yiyixuxu

**Comment by foreverpiano** [comment]
@DN6 #10033
Has this been fixed now?

**Comment by SahilCarterr** [comment]
#### Official Diffusers Code
https://github.com/user-attachments/assets/5397536c-6177-4193-a773-8a33d444e767
#### PR #10033 Close to Merged
https://github.com/user-attachments/assets/02d697c6-d699-414d-8613-f252b8948cdb

Quality has much improved , now fixed @foreverpiano

**Comment by hlky** [comment]
@jmahajan117 Install from `main`. We are preparing for 0.32 release in the next weeks.

**Comment by foreverpiano** [comment]
I think the issue is still not fixed. The Mochi attention does not consider adding masks (which are needed for text padding). So the result is different from the original Mochi.

The encoder query should have an attention mask because it is padded to 256. Padding with zeros is not equivalent to adding an attention mask when considering the softmax operation in attention.

https://github.com/huggingface/diffusers/blob/63243406ba5510c10d5cac931882918ceba926f9/src/diffusers/models/attention_processor.py#L3928

cc @hlky @DN6 @a-r-r-o-w

**Comment by a-r-r-o-w** [comment]
@foreverpiano Could you use this branch? https://github.com/huggingface/diffusers/tree/mochi-quality

https://github.com/huggingface/diffusers/blob/09fe7ecd26eec83fc14fe77e8d767ed45a71176e/src/diffusers/models/transformers/transformer_mochi.py#L278

**Comment by foreverpiano** [comment]
@a-r-r-o-w The implementation appears better since it takes the mask into consideration. I believe this addresses my previous concern. Let me look into this version video output.
